### ISSUE TITLE
Did you try less blocks for C- Abstractor and D-Abstractor

### ISSUE THREAD

**Issue by laserwave** [kakaobrain/honeybee#5]
As is mentioned in your paper, you use six blocks in both C- Abstractor and D-Abstractor. Did you try less blocks?

> we utilize total six blocks in both C- Abstractor and D-Abstractor (i.e., L = 3 for C-Abstractor and L = 6 for D-Abstractor in Fig. 4)

**Comment by laserwave** [comment]
And how many layers are used in the mlp in C-Abstractor?

**Comment by khanrc** [comment]
4-6 blocks have worked well in our experiments. C-Abstractor uses a 2-layer MLP (1 hidden layer).

**Comment by laserwave** [comment]
Did you find it more difficult to achieve convergence for c-abstractor or d-abstractor than simple 2-layer mlp(number of visual tokens almost the same)? Maybe you train it for more iterations than the mlp or linear projector?

**Comment by khanrc** [comment]
We used the same iterations across all architectures in comparison: 50k for pre-training and 4k for instruction tuning. The early convergence is 2-layer MLP > C-Abstractor (in terms of pre-training loss), but the loss of C-Abstractor becomes lower than 2-layer MLP after 10k iterations.

**Comment by laserwave** [comment]
> We used the same iterations across all architectures in comparison: 50k for pre-training and 4k for instruction tuning. The early convergence is 2-layer MLP > C-Abstractor (in terms of pre-training loss), but the loss of C-Abstractor becomes lower than 2-layer MLP after 10k iterations.

That's weird, the pre-training loss of 2-layer MLP is always smaller than that of C-Abstractor during one epoch(about 460w training instances) pretraining phase in my experiment. What is the output hidden size of the two reg blocks in the C-Abstractor? Same as vision encoder or same as llm? Maybe this value should not be set to a large value.

**Comment by khanrc** [comment]
What is the meaning of 460w training instances? We use the same hidden size of reg blocks with the vision encoder, and it is matched with LLM in MLP.

**Comment by laserwave** [comment]
> What is the meaning of 460w training instances? We use the same hidden size of reg blocks with the vision encoder, and it is matched with LLM in MLP.

Get. It means 4,600,000 train samples.
